src/mca/plm/base/plm_base_receive.c

```diff
@@ -171,7 +172,7 @@ void prte_plm_base_recv(int status, prte_process_name_t* sender,
         if (PRTE_SUCCESS == rc) {
             job = jb.jobid;
         }
-        PRTE_DESTRUCT(&jb);
+        // Keep the object in case we need it for reference in the prte_job_data hash
```

Comment:
I think what's going on here is that the jb object got stored in the job hash table since this is the first instance of the jobid/namespace. maybe the comment would be clearer stating that the jb structure cannot be released since it may have been added to the job hash table.

Reply:
I'll cleanup the comment.